lib/modules/asset/AssetsController.ts

```diff
@@ -230,17 +252,54 @@ export class AssetsController {
     const sort = request.input.body?.sort;
     const type = request.input.args.type;
 
-    const exporter = new MeasureExporter(this.plugin, engineId, {
-      target: "asset",
-    });
+    const link = await this.measureExporter.prepareExport(
+      engineId,
+      request.getUser(),
+      {
+        endAt,
+        id,
+        query,
+        sort,
+        startAt,
+        type,
+      }
+    );
 
-    const { link } = await exporter.prepareExport(request.getUser(), id, {
-      endAt,
-      query,
-      sort,
-      startAt,
-      type,
-    });
+    return { link };
+  }
+
+  async export(request: KuzzleRequest) {
+    const engineId = request.getString("engineId");
+
+    if (
+      request.context.connection.protocol === "http" &&
+      request.context.connection.misc.verb === "GET"
+    ) {
+      const exportId = request.getString("exportId");
+
+      request.response.configure({
+        headers: {
+          "Content-Disposition": `attachment; filename="${InternalCollection.ASSETS}.csv"`,
+          "Content-Type": "text/csv",
+        },
+      });
+
+      const stream = await this.exporter.sendExport(engineId, exportId);
+
+      return new HttpStream(stream);
+    }
+
+    const query = request.input.body?.query;
+    const sort = request.input.body?.sort;
```

Comment:
```
const query = request.getBodyObject("query");
const sort = request.getBodyString("sort");
```

In order to correctly handle missing properties ?

Reply:
`query` and `sort` are objects and not mandatory.
This method allow optional properties while  `getBodyObject("query")` define it mandatory.
But it's true refactor it can be better, but need improvement in `kuzzle-sdk` on in `core`.

Reply:
my bad, misunderstood the code :+1: